src/search/FindInFiles.js

```diff
@@ -417,15 +418,26 @@ define(function (require, exports, module) {
         }
         
         var scopeName = searchModel.scope ? searchModel.scope.fullPath : ProjectManager.getProjectRoot().fullPath,
-            perfTimer = PerfUtils.markStart("FindIn: " + scopeName + " - " + queryInfo.query);
+            perfTimer = PerfUtils.markStart("FindIn: " + scopeName + " - " + queryInfo.query),
+            searchDeferred = new $.Deferred();
         
-        return candidateFilesPromise
+        candidateFilesPromise
             .then(function (fileListResult) {
                 // Filter out files/folders that match user's current exclusion filter
                 fileListResult = FileFilters.filterFileList(filter, fileListResult);
+                fileListResult = FindUtils.getSortedFiles(fileListResult, function (file) { return file.fullPath; }, FindUtils.getOpenFilePath());
```

Comment:
The list comes from `ProjectModel._getAllFilesCache()`. It should be easy to change `allFilesVisitor()` function to insert into sorted list (currently just to end), but I'm sure that will cause a slowdown somewhere else.

I also see that `_resetCache()` is called *every* time `ProjectModel.handleFSEvent()` is called, which happens on every write to `state.json`, which happens on every click in file tree. So, that may need to be made smarter about when to clear cache.